Integrations/integration-Cymon.yml

```diff
@@ -0,0 +1,543 @@
+
+    DEFAULT_HEADERS = {
+      "Content-Type": "application/json"
+    }
+
+    ''' HELPER FUNCTIONS '''
+
+    def http_request(method, url, headers):
+        try:
+            res = requests.request(method,
+                                    url,
+                                    verify=USE_SSL,
+                                    headers=headers)
+
+            if res.status_code == 200:
+                return res.json()
+            # 204 HTTP status code is returned when api rate limit has been exceeded
+            elif res.status_code == 204:
+                return_error("You've reached your API call quota.")
+            elif res.status_code == 404:
+                return {}
+
+            res.raise_for_status()
+
+        except Exception, e:
+            raise(e)
+
+    ''' DOMAIN COMMAND '''
+    def get_domain_full_report(domain):
+        report_results = []
+
+        from_param = 0
+        size_param = 10
+        total = None
+
+        url = '{}/{}/{}?from={}&size={}'.format(SERVER_URL_V2, 'domain', domain, from_param, size_param)
+
+        while total is None or total > from_param:
+            response = http_request('GET', url, DEFAULT_HEADERS)
+
+            hits = response.get('hits', [])
+            for hit in hits:
+                timestamp = datetime.datetime.strptime(hit.get('timestamp'), '%Y-%m-%dT%H:%M:%S.%fZ')
+
+                report_results.append({
+                    'Title': hit.get('title').title(),
+                    'Feed': hit.get('feed'),
+                    'Timestamp': timestamp.strftime("%Y-%m-%d %H:%M:%S"),
+                    'Tags': hit.get('tags'),
+                    'Hostname': hit.get('ioc', {}).get('hostname'),
+                    'IP': hit.get('ioc', {}).get('ip'),
+                    'Domain': hit.get('ioc', {}).get('domain'),
+                    'Reported By': hit.get('reported_by'),
+                    'Location': hit.get('location', {}).get('country')
+                })
+
+            from_param = from_param + size_param
+            total = int(response.get('total', 0))
+
+            url = '{}/{}/{}?from={}&size={}'.format(SERVER_URL_V2, 'domain', domain, from_param, size_param)
+
+        return report_results
+
+    def get_domain_report(domain):
+        reports = {}
+
+        from_param = 0
+        size_param = 10
+        total = None
+
+        url = '{}/{}/{}?from={}&size={}'.format(SERVER_URL_V2, 'domain', domain, from_param, size_param)
+
+        while total is None or total > from_param:
+            response = http_request('GET', url, DEFAULT_HEADERS)
+
+            hits = response.get('hits', [])
+            for hit in hits:
+                title = hit.get('title')
+                timestamp = datetime.datetime.strptime(hit.get('timestamp'), '%Y-%m-%dT%H:%M:%S.%fZ')
+
+                if (title in reports and reports.get(title).get('timestamp') < timestamp) or title not in reports:
```

Comment:
@MarkShleifer isn't `reports.get(title).get('timestamp')` a string and `timestamp` a `datetime` object ? if so, does the comparison works properly?

Reply:
reports.get(title).get('timestamp') is a timestamp, because it was set saved in previous iterations